packages/admin-panel/src/VizBuilderApp/api/queries/useEntities.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import { useQuery } from 'react-query';
+import { stringifyQuery } from '@tupaia/utils';
+import { get } from '../api';
+import { DEFAULT_REACT_QUERY_OPTIONS } from '../constants';
+
+export const useEntities = search =>
+  useQuery(
+    ['entities'],
+    async () => {
+      const endpoint = stringifyQuery(undefined, `entities`, {
+        columns: JSON.stringify(['name', 'code']),
+        filter: JSON.stringify({
+          name: { comparator: 'ilike', comparisonValue: `%${search}%`, castAs: 'text' },
+        }),
+      });
+      return get(endpoint);
+    },
+    {
+      ...DEFAULT_REACT_QUERY_OPTIONS,
```

Comment:
Can you add a debounce here? I haven't tried but I guess it would be a bit lagging when adding words.

Reply:
Ah great catch @biaoli0! I was caching the query and not refetching based on the search term. I think it should be fixed now 👍